### PR TITLE
Drop python 3.6 support from dependencies in setup.py

### PR DESCRIPTION
**Pull request by sanjanag**
Closes https://github.com/cleanlab/cleanlab/issues/576

### CODE REVIEW

**Comment by sanjanag** [comment]
Looked up release history of packages to check the python versions supported.

- Supports >=3.6.1 until [1.1.5](https://pypi.org/project/pandas/1.1.5/), and >=3.7.1 from  [1.2.0](https://pypi.org/project/pandas/1.2.0/)
- Latest [version](https://pypi.org/project/tqdm/4.64.1/) also supports >=2.7, hence did not change the version.
- Numpy 3.6 support dropped in [this](https://numpy.org/doc/stable/release/1.20.0-notes.html) release
- Supported 3.x until [1.1.0](https://pypi.org/project/termcolor/1.1.0/), supporting >=3.7 from [2.0.0](https://pypi.org/project/termcolor/2.0.0/)

**Comment by jwmueller** [comment]
Any idea why pandas 1.2.0 does not support python 3.7.0 but only > 3.7.1? Should we stick with pandas >= 1.1.5 instead?

File: setup.py [hunk 1]
```diff
@@ -89,10 +88,10 @@ def run(self):
...
     # https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/
     install_requires=[
-        "numpy>=1.11.3",
-        "scikit-learn>=0.18",
+        "numpy>=1.20.0",
+        "scikit-learn>=1.0",
         "tqdm>=4.53.0",
-        "pandas>=1.0.0",
-        "termcolor>=1.1.0",
+        "pandas>=1.2.0",
```
Comment (jwmueller) [inline setup.py:94]:
```suggestion
        "pandas>=1.1.5",
```

Comment (jwmueller) [reply setup.py:94]: Should we do this ^ instead?  Since pandas 1.2.0 does not support python 3.7.0 but only > 3.7.1; I'm not sure why that is?

**Review by jwmueller — APPROVED** [review]
Lgtm

**Pull request merged**